src/Nethermind/Nethermind.Network/P2P/MessageDictionary.cs

```diff
@@ -55,7 +56,7 @@ public void Send(Request<T66Msg, TData> request)
     {
         if (_requestCount >= MaxConcurrentRequest)
         {
-            throw new InvalidOperationException("Concurrent request limit reached");
+            throw new ConcurrencyLimitReachedException($"Concurrent request limit reached. Message type: {typeof(TMsg)}");
```

Comment:
Btw this limit is per message? Per Peer?

Reply:
Per peer.